### PR TITLE
[Feature] show create materialized view

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close (https://github.com/apache/incubator-doris/issues/9196)

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by hf200012 — DISMISSED** [review]
LGTM

**Review by hf200012 — DISMISSED** [review]

**Comment by qidaye** [comment]
It's better to add a `show create materialized view` doc pointer to the materialized view document

**Comment by stalary** [comment]
> It's better to add a `show create materialized view` doc pointer to the materialized view document

Thanks，I forgot that, and I'm going to add it

**Comment by stalary** [comment]
> It's better to add a `show create materialized view` doc pointer to the materialized view document

done

**Comment by qidaye** [comment]
Thanks for your quick fix.
> It's better to add a `show create materialized view` doc pointer to the materialized view document

What I mean is to add a note in the materialized view document showing the creation statement, like drop
https://github.com/apache/incubator-doris/blob/98bfeaf560f7ac04f2324499e4dc7cc3bbf0a437/docs/en/advanced/materialized-view.md?plain=1#L147-L155

**Review by hf200012 — DISMISSED** [review]

**Comment by stalary** [comment]
> Thanks for your quick fix.
> 
> > It's better to add a `show create materialized view` doc pointer to the materialized view document
> 
> What I mean is to add a note in the materialized view document showing the creation statement, like drop
> 
> https://github.com/apache/incubator-doris/blob/98bfeaf560f7ac04f2324499e4dc7cc3bbf0a437/docs/en/advanced/materialized-view.md?plain=1#L147-L155

done

File: docs/en/advanced/materialized-view.md [hunk 1]
```diff
@@ -148,11 +148,13 @@ You can see that the current `mv_test` table has three materialized views: mv\_1
 
 If the user no longer needs the materialized view, you can delete the materialized view by 'DROP' commen.
 
-The specific syntax can be viewed through the following command:
+You can view the specific syntax[SHOW CREATE MATERIALIZED VIEW](../sql-manual/sql-reference/Data-Definition-Statements/Drop/DROP-MATERIALIZED-VIEW.html)
```
Comment (qidaye) [inline docs/en/advanced/materialized-view.md:151]:
```suggestion
You can view the specific syntax[SHOW CREATE MATERIALIZED VIEW](../sql-manual/sql-reference/Data-Definition-Statements/Drop/DROP-MATERIALIZED-VIEW.md)
```

File: docs/en/advanced/materialized-view.md [hunk 2]
```diff
@@ -148,11 +148,13 @@ You can see that the current `mv_test` table has three materialized views: mv\_1
...
 
-```
-HELP DROP MATERIALIZED VIEW
-```
+### View the materialized view that has been created
+
+Users can view the created materialized views by using commands
+
+You can view the specific syntax[SHOW CREATE MATERIALIZED VIEW](../sql-manual/sql-reference/Show-Statements/SHOW-CREATE-MATERIALIZED-VIEW.html)
```
Comment (qidaye) [inline docs/en/advanced/materialized-view.md:157]:
```suggestion
You can view the specific syntax[SHOW CREATE MATERIALIZED VIEW](../sql-manual/sql-reference/Show-Statements/SHOW-CREATE-MATERIALIZED-VIEW.md)
```

File: docs/zh-CN/advanced/materialized-view.md [hunk 3]
```diff
@@ -146,7 +146,11 @@ MySQL [test]> desc mv_test all;
 
 具体的语法可查看[DROP MATERIALIZED VIEW](../sql-manual/sql-reference/Data-Definition-Statements/Drop/DROP-MATERIALIZED-VIEW.html)
```
Comment (qidaye) [inline docs/zh-CN/advanced/materialized-view.md:147]:
```suggestion
具体的语法可查看[DROP MATERIALIZED VIEW](../sql-manual/sql-reference/Data-Definition-Statements/Drop/DROP-MATERIALIZED-VIEW.md) 
```

File: docs/zh-CN/advanced/materialized-view.md [hunk 4]
```diff
@@ -146,7 +146,11 @@ MySQL [test]> desc mv_test all;
 
 具体的语法可查看[DROP MATERIALIZED VIEW](../sql-manual/sql-reference/Data-Definition-Statements/Drop/DROP-MATERIALIZED-VIEW.html) 
 
+### 查看已创建的物化视图
 
+用户可以通过命令查看已创建的物化视图的
+
+具体的语法可查看[SHOW CREATE MATERIALIZED VIEW](../sql-manual/sql-reference/Show-Statements/SHOW-CREATE-MATERIALIZED-VIEW.html)
```
Comment (qidaye) [inline docs/zh-CN/advanced/materialized-view.md:153]:
```suggestion
具体的语法可查看[SHOW CREATE MATERIALIZED VIEW](../sql-manual/sql-reference/Show-Statements/SHOW-CREATE-MATERIALIZED-VIEW.md)
```

**Review by qidaye — COMMENTED** [review]
Better with `md` file.

**Comment by stalary** [comment]
> Better with `md` file.

Thanks. done

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 5]
```diff
@@ -2178,4 +2182,25 @@ private void handleAdminDiagnoseTablet() {
         resultSet = new ShowResultSet(showMetaData, resultRowSet);
     }
 
+    private void handleShowCreateMaterializedView() throws AnalysisException {
+        List<List<String>> resultRowSet = new ArrayList<>();
+        ShowCreateMaterializedViewStmt showStmt = (ShowCreateMaterializedViewStmt) stmt;
+        Database db = Catalog.getCurrentCatalog().getDbOrAnalysisException(showStmt.getTableName().getDb());
+        Table table = db.getTableOrAnalysisException(showStmt.getTableName().getTbl());
+        OlapTable baseTable = ((OlapTable) table);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:2190]: should check table type before cast

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 6]
```diff
@@ -2178,4 +2182,25 @@ private void handleAdminDiagnoseTablet() {
...
+        Long indexIdByName = baseTable.getIndexIdByName(showStmt.getMvName());
+        MaterializedIndexMeta meta = baseTable.getIndexMetaByIndexId(indexIdByName);
+        if (meta == null || meta.getDefineStmt() == null) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:2193]:
```suggestion
        if (meta != null && meta.getDefineStmt() != null) {
            String originStmt = meta.getDefineStmt().originStmt;
            List<String> data = new ArrayList<>();
            data.add(showStmt.getTableName().getTbl());
            data.add(showStmt.getMvName());
            data.add(originStmt);
            resultRowSet.add(data);
        }
        resultSet = new ShowResultSet(showStmt.getMetaData(), resultRowSet);
```

**Comment by stalary** [comment]
done. @morrySnow

**Comment by qidaye** [comment]
LGTM
@stalary please resolve confilicts.

**Comment by stalary** [comment]
> LGTM @stalary please resolve confilicts.

done.

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**